tests/js/spec/views/performance/trends/index.spec.tsx

```diff
@@ -17,42 +31,125 @@ const trendsViewQuery = {
   query: `tpm():>0.01 transaction.duration:>0 transaction.duration:<${DEFAULT_MAX_DURATION}`,
 };
 
+jest.mock(
+  'sentry/utils/getDynamicComponent',
+  () =>
+    ({fixed}) =>
+      fixed
+);
+
 jest.mock('moment', () => {
   const moment = jest.requireActual('moment');
   moment.now = jest.fn().mockReturnValue(1601251200000);
   return moment;
 });
 
-function selectTrendFunction(wrapper, field) {
-  const menu = wrapper.find('TrendsDropdown DropdownMenu');
-  expect(menu).toHaveLength(2);
-  menu.find('DropdownButton').first().simulate('click');
+async function getTrendDropdown() {
+  const dropdown = (await screen.findAllByTestId('dropdown-control'))[0];
+  expect(dropdown).toBeInTheDocument();
+  return dropdown;
+}
+
+async function getParameterDropdown() {
+  const dropdown = (await screen.findAllByTestId('dropdown-control'))[1];
+  expect(dropdown).toBeInTheDocument();
+  return dropdown;
+}
+
+async function selectMenuItemFromDropdown(dropdownEl, testid) {
+  (browserHistory.push as any).mockReset();
+  expect(browserHistory.push).not.toHaveBeenCalled();
+  const option = await within(await within(dropdownEl).findByTestId(testid)).findByTestId(
+    'menu-item'
+  );
+  expect(option).toBeInTheDocument();
+  clickEl(option);
+
+  await waitFor(() => expect(browserHistory.push).toHaveBeenCalled());
+}
+
+async function selectTrendFunction(field) {
+  const dropdown = await getTrendDropdown();
+  clickEl(dropdown);
+
+  await selectMenuItemFromDropdown(dropdown, field);
+}
+
+async function getDropdownButtons() {
+  return await screen.findAllByTestId('dropdown-control-button');
+}
 
-  const option = menu.find(`DropdownItem[data-test-id="${field}"] span`);
-  expect(option).toHaveLength(1);
-  option.simulate('click');
+async function selectTrendParameter(label) {
+  const dropdown = await getParameterDropdown();
+  clickEl(dropdown);
 
-  wrapper.update();
+  await selectMenuItemFromDropdown(dropdown, label);
 }
 
-function selectTrendParameter(wrapper, label) {
-  const menu = wrapper.find('TrendsDropdown DropdownMenu');
-  expect(menu).toHaveLength(2);
-  menu.find('DropdownButton').at(1).simulate('click');
+async function waitForMockCall(mock: any) {
+  await waitFor(() => {
+    expect(mock).toHaveBeenCalled();
+  });
+}
 
-  const option = menu.find(`DropdownItem[data-test-id="${label}"] span`);
-  expect(option).toHaveLength(1);
-  option.simulate('click');
+async function enterSearch(el, text) {
+  fireEvent.change(el, {target: {value: text}});
+  fireEvent.submit(el);
+}
+
+// Might swap on/off the skiphover to check perf later.
+async function clickEl(el) {
+  userEvent.click(el, undefined, {skipHover: true, skipPointerEventsCheck: true});
```

Comment:
Should this be `fireEvent.click` instead? It looks like you're turning off most of what makes `userEvent.click` different from `fireEvent.click`

Reply:
Yes, but the nice part of having this indirect function is that I can easily change it for the whole file later. 🤷

Reply:
I'm going to put this out for now and see what the performance of it is.